### PR TITLE
Make `Option.()` compatible with ruby >= 2.7

### PR DESCRIPTION
**Pull request by yogeshjain999**

### CODE REVIEW

File: test/option_test.rb [hunk 1]
```diff
@@ -7,12 +7,12 @@ def Option(*args)
...
   # proc
-  it { Option( ->(*args) { "proc! #{args.inspect}" } ).(1,2).must_equal "proc! [1, 2]" }
+  it { Option( ->(*args, **) { "proc! #{args.inspect}" } ).(1,2).must_equal "proc! [1, 2]" }
   it { Option( lambda { "proc!" } ).().must_equal "proc!" }
 
   # proc with instance_exec
-  it { Option( ->(*args) { "#{self.class} #{args.inspect}" } ).(Object, 1, 2).must_equal "OptionTest [Object, 1, 2]" }
-  it { Option( ->(*args) { "#{self} #{args.inspect}" }, instance_exec: true ).(Object, 1, 2).must_equal "Object [1, 2]" }
+  it { Option( ->(*args, **) { "#{self.class} #{args.inspect}" } ).(Object, 1, 2).must_equal "OptionTest [Object, 1, 2]" }
+  it { Option( ->(*args, **) { "#{self} #{args.inspect}" }, instance_exec: true ).(Object, 1, 2).must_equal "Object [1, 2]" }
```
Comment (apotonick) [inline test/option_test.rb]: Do we need the `, **`?

Comment (yogeshjain999) [reply test/option_test.rb]: Yes. By not mentioning it, Option gives different results for `args.inspect` between ruby <= 2.7 and  ruby 3.

```ruby
->(*args) { "#{args.inspect}" }.  # ruby < 2.6 => "Object [1, 2, {}]"
->(*args) { "#{args.inspect}" }.  # ruby 3 => "Object [1, 2]"
```

File: test/option_test.rb [hunk 2]
```diff
@@ -12,7 +12,8 @@ def Option(*args)
 
   # proc with instance_exec
   it { Option( ->(*args) { "#{self.class} #{args.inspect}" } ).(Object, 1, 2).must_equal "OptionTest [Object, 1, 2]" }
-  it { Option( ->(*args) { "#{self} #{args.inspect}" }, instance_exec: true ).(Object, 1, 2).must_equal "Object [1, 2]" }
+  it { Option( ->(*args, **) { "#{self} #{args.inspect}" }, instance_exec: true ).(Object, 1, 2).must_equal "Object [1, 2]" }
+  it { Option( ->(*args, a:, b:, **kwargs) { "#{self} #{args.inspect} #{a} #{b} #{kwargs}" }, instance_exec: true ).(Object, 1, 2, keyword_arguments: { a: 3, b: 4, c: 5 }).must_equal "Object [1, 2] 3 4 {:c=>5}" }
```
Comment (yogeshjain999) [inline test/option_test.rb]: I've added this test for explicitness.

File: test/option_test.rb [hunk 3]
```diff
@@ -7,26 +7,29 @@ def Option(*args)
...
-  it { Option( ->(*args) { "proc! #{args.inspect}" } ).(1,2).must_equal "proc! [1, 2]" }
-  it { Option( lambda { "proc!" } ).().must_equal "proc!" }
+  it { _(Option( ->(*args) { "proc! #{args.inspect}" } ).(1,2)).must_equal "proc! [1, 2]" }
+  it { _(Option( lambda { "proc!" } ).()).must_equal "proc!" }
 
   # proc with instance_exec
-  it { Option( ->(*args) { "#{self.class} #{args.inspect}" } ).(Object, 1, 2).must_equal "OptionTest [Object, 1, 2]" }
-  it { Option( ->(*args) { "#{self} #{args.inspect}" }, instance_exec: true ).(Object, 1, 2).must_equal "Object [1, 2]" }
+  it { _(Option( ->(*args) { "#{self.class} #{args.inspect}" } ).(Object, 1, 2)).must_equal "OptionTest [Object, 1, 2]" }
+  it { _(Option( ->(*args, **kwargs) { "#{self} #{args.inspect} #{kwargs.inspect}" }, instance_exec: true ).(Object, 1, 2)).must_equal "Object [1, 2] {}" }
```
Comment (apotonick) [inline test/option_test.rb]: Does this test :point_up: fail without `, **kwargs`?

Comment (yogeshjain999) [reply test/option_test.rb]: Nope, added new test without `**kwargs` and `instance_exec : true`, line no - 15

**Review by apotonick — APPROVED** [review]

**Pull request merged**